app/api/users.v2/model/User.ts

```diff
@@ -17,3 +17,6 @@ export class User {
     return ['admin', 'editor'].includes(this.role);
   }
 }
+
+export type { UserRole };
```

Comment:
Particularly in the case of the domain models, I think it's usually more readable and modular to instead of exporting internal types, just reference them relative to the model. In this case `User['role']`.
But at this point it's just an opinion :)